lib/Cake/Model/Datasource/Session/DatabaseSession.php

```diff
@@ -114,7 +114,17 @@ public function write($id, $data) {
 		$expires = time() + $this->_timeout;
 		$record = compact('id', 'data', 'expires');
 		$record[$this->_model->primaryKey] = $id;
-		return $this->_model->save($record);
+
+		$options = array(
+			'validate' => false,
+			'callbacks' => false,
+			'counterCache' => false
+		);
+		try {
+			return $this->_model->save($record, $options);
+		} catch (Exception $e) {
+			return $this->_model->save($record, $options);
```

Comment:
If it fails again, it's because of a none-transient problem and that'll bubble up (rightly so IMO).

I see I left it catching all exceptions, I'll update that to PDOException since that's the only scenario that this code is expecting to encounter.